### PR TITLE
Print spinners during backend interaction

### PR DESCRIPTION
**Pull request by nbrahms**
Currently, we provide no user feedback whilst we contact the P0 backend. This makes it difficult for the user to distinguish between waiting on a backend response and a hung process.

Use a yarn-style Unicode spinner to indicate that the process is still live.

https://github.com/user-attachments/assets/bd58acc0-7d1a-4058-a41a-d95c3c13a929

### CODE REVIEW

**Review by komal-dhull — APPROVED** [review]
looks good, thanks!

**Comment by GGonryun** [comment]
This is coooooool! 🥳

**Review by GGonryun — APPROVED** [review]

File: src/drivers/stdio.ts [hunk 1]
```diff
@@ -39,7 +40,50 @@ export function print2(message: any) {
...
+const Spin = {
+  items: ["⠇", "⠋", "⠙", "⠸", "⠴", "⠦"],
+  delayMs: 200,
+};
+
+/** Prints a Unicode spinner until a promise resolves */
+export const spinUntil = async <T>(message: string, promise: Promise<T>) => {
+  let isDone = false;
+  let ix = 0;
+  void promise.finally(() => (isDone = true)).catch(() => {});
```
Comment (fabgo) [inline src/drivers/stdio.ts:73]: Should we swallow errors here?

**Review by fabgo — APPROVED** [review]

File: src/drivers/stdio.ts (hunk 1 above)
Comment (nbrahms) [reply src/drivers/stdio.ts:62]: The errors are thrown on line 77

**Pull request merged**